### PR TITLE
Uses router state to handle provider request success handling

### PR DESCRIPTION
**Pull request by cheshireoctopus**
## Overview

Instead of setting a query param when redirecting providers from the provider form to the provider success page, this PR uses `state` within the browser history.

### CODE REVIEW

File: app/javascript/packs/pages/provider_signup/SignupStepper.jsx [hunk 1]
```diff
@@ -137,7 +137,17 @@ export default function SignupStepper({ steps }) {
...
+      <Redirect
+        push
+        to={{
+          pathname: `/providers/${redirectId}`,
+          state: {
+            providerCreation: true,
+          },
+        }}
+      />
+    )
```
Comment (cheshireoctopus) [inline app/javascript/packs/pages/provider_signup/SignupStepper.jsx]: Please note that this method adds this attribute to the browser state.

The user will either have to replace the state (open a new window, navigate to new URL) in order for the state to clear.

Other users will not be impacted.

File: app/javascript/packs/pages/provider_signup/SignupStepper.jsx [hunk 2]
```diff
@@ -137,7 +137,17 @@ export default function SignupStepper({ steps }) {
   const CurrentStep = steps[activeStep] && steps[activeStep].component;
 
   if (redirectId) {
-    return <Redirect to={`/providers/${redirectId}?success`} />
+    return (
+      <Redirect
+        push
+        to={{
+          pathname: `/providers/${redirectId}`,
```
Comment (tuntisz) [inline app/javascript/packs/pages/provider_signup/SignupStepper.jsx:118]: what are your thoughts on making a route like `/provider-signup/success` and passing the provider ID into that? My thinking is that using this route for confirmation and provider viewing could make traffic data harder to understand if we were to add something like google analytics

File: app/javascript/packs/pages/provider_signup/SignupStepper.jsx (hunk 1 above)
Comment (tuntisz) [reply app/javascript/packs/pages/provider_signup/SignupStepper.jsx]: Awesome! I didn't know this existed

File: app/javascript/packs/pages/provider_signup/SignupStepper.jsx (hunk 2 above)
Comment (cheshireoctopus) [reply app/javascript/packs/pages/provider_signup/SignupStepper.jsx:118]: Oooh. Good points.

I recall the initial concern about passing the query param `?success` to be about providers copying that URL and sharing it, thus resulting in page where users could not view the "Offer Help" button.

Does the `/provider-signup/success` route put us in the same position?

Comment (tuntisz) [reply app/javascript/packs/pages/provider_signup/SignupStepper.jsx:118]: Ah, yes. I suppose it does. I'm over-thinking, this work is great!

Tangentially related, it seems like we should add UI on the success that says "Share this link with your friends".  I'll chat with folks.

**Review by cantino — APPROVED** [review]
Nice!

File: app/javascript/packs/pages/provider_signup/SignupStepper.jsx (hunk 2 above)
Comment (cheshireoctopus) [reply app/javascript/packs/pages/provider_signup/SignupStepper.jsx:118]: We have a spec for it, but do not believe we have discussed URL shortening?

**Pull request merged**